bleak/backends/corebluetooth/CentralManagerDelegate.py

```diff
@@ -146,11 +147,11 @@ async def connect_(self, peripheral: CBPeripheral, timeout=10.0) -> bool:
         self._connection_state = CMDConnectionState.PENDING
         self.central_manager.connectPeripheral_options_(peripheral, None)
 
-        start = time.time()
-        while self._connection_state == CMDConnectionState.PENDING:
-            await asyncio.sleep(0)
-            if time.time() - start > timeout:
-                raise TimeoutError("Failed to connect in %d seconds" % timeout)
+        try:
```

Comment:
I think we need to call `self._connection_state_changed.clear()` here in case connect is called more than once.

Reply:
Great idea.